pkg/list/values/extract.go

```diff
@@ -0,0 +1,308 @@
+)
+
+// handleSpecialComponent processes special components like settings and metadata.
+func handleSpecialComponent(stack map[string]interface{}, component string) (map[string]interface{}, bool) {
+	// First check if the component exists at the top level
+	if section, ok := stack[component].(map[string]interface{}); ok {
+		return section, true
+	}
+
+	// If not found at the top level and component is "settings", look for it in components
+	if component == "settings" {
+		return extractSettingsFromComponents(stack)
+	}
+
+	return nil, false
+}
+
+// extractSettingsFromComponents extracts settings from terraform components.
+func extractSettingsFromComponents(stack map[string]interface{}) (map[string]interface{}, bool) {
+	allSettings := make(map[string]interface{})
+
+	// Try to navigate to terraform components
+	components, ok := stack["components"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	terraform, ok := components["terraform"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	// Collect settings from all terraform components
+	for componentName, componentData := range terraform {
+		if settings := extractComponentSettings(componentData); settings != nil {
+			allSettings[componentName] = settings
+		}
+	}
+
+	// Return all settings if we found any
+	if len(allSettings) > 0 {
+		return allSettings, true
+	}
+
+	return nil, false
+}
+
+// extractComponentSettings extracts settings from a component.
+func extractComponentSettings(componentData interface{}) interface{} {
+	comp, ok := componentData.(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	settings, ok := comp["settings"].(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	return deepCopyToStringMap(settings)
+}
+
+// deepCopyToStringMap creates a deep copy of a map, ensuring all keys are strings.
+// This helps with JSON marshaling which requires string keys.
+func deepCopyToStringMap(m interface{}) interface{} {
+	switch m := m.(type) {
+	case map[string]interface{}:
+		copy := make(map[string]interface{})
+		for k, v := range m {
+			copy[k] = deepCopyToStringMap(v)
+		}
+		return copy
+	case map[interface{}]interface{}:
+		copy := make(map[string]interface{})
+		for k, v := range m {
+			copy[fmt.Sprintf("%v", k)] = deepCopyToStringMap(v)
+		}
+		return copy
+	case []interface{}:
+		copy := make([]interface{}, len(m))
+		for i, v := range m {
+			copy[i] = deepCopyToStringMap(v)
+		}
+		return copy
+	default:
+		return m
+	}
+}
+
+// handleTerraformComponent processes regular terraform components.
+func handleTerraformComponent(stack map[string]interface{}, component string, includeAbstract bool) (map[string]interface{}, bool) {
+	components, ok := stack["components"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	terraform, ok := components["terraform"].(map[string]interface{})
+	if !ok {
+		return nil, false
+	}
+
+	componentName := strings.TrimPrefix(component, "terraform/")
```

Comment:
I'm concerned this is a heuristic that assumes that terraform components are in a folder like `components/terraform/` but that's driven by configuration.